webapp/src/components/systemConsole/teamFilter/teamIcon/teamIcon.tsx

```diff
@@ -0,0 +1,51 @@
+// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
+// See LICENSE.txt for license information.
+
+import React from 'react';
+
+import type {Team} from '@mattermost/types/teams';
+
+import {Client4} from 'mattermost-redux/client';
+
+import type {DropdownOption} from 'components/common/dropdown/dropdown';
+
+import './style.scss';
+
+export type Props = {
+    data: DropdownOption;
+}
+function TeamIcon({data}: Props) {
+    // @ts-expect-error last_team_icon_update
+    // field will be included in latest types release,
+    // but its there in the data already
+    if (data.raw.last_team_icon_update) {
+        // last_team_icon_update is set whenever a team icon is set or updated.
+        // So, a truthy value of `last_team_icon_update` field indicates precense of
+        // team icons, so here we use it to show the team icon image.
+
+        // @ts-expect-error sane reason as above about the last_team_icon_update field
+        const teamURL = Client4.getTeamIconUrl(data.value, (data.raw as Team)?.last_team_icon_update);
```

Comment:
We can omit most of this if we just check for `last_team_icon_update` to be in the raw. Something like this:

```suggestion
    if (`last_team_icon_update` in data.raw && typeof(data.raw.last_team_icon_update) === 'number' && data.raw.last_team_icon_update) {
        const teamURL = Client4.getTeamIconUrl(data.value, data.raw.last_team_icon_update);
```

Reply:
I updated the `@mattermost/types` package to bring in the `last_team_icon_update` field, so no more need of all this checks. The field now exists in type definition.